rclcpp/src/rclcpp/duration.cpp

```diff
@@ -246,6 +246,26 @@ Duration::to_rmw_time() const
   return result;
 }
 
+Duration
+Duration::from_rmw_time(rmw_time_t duration)
```

Comment:
Why does the logic here differ so much from `Duration::Duration(const builtin_interfaces::msg::Duration & duration_msg)`? It seems like they do the same thing in principle.

Reply:
The logic is the "same", but considering that in `builtin_interfaces::msg::Duration` seconds is an `int32_t` and nanoseconds an `uint32_t`, the result represented as nanoseconds always fits in an `int64_t` (`rcl_duration_value_t`).

Whereas `rmw_time_t` seconds and nanoseconds are both a `uint64_t`, and the result in nanoseconds might not fit in `int64_t`.

Reply:
Makes sense. 👍